src/macros-by-example.md

```diff
@@ -193,6 +193,76 @@ compiler knows how to expand them properly:
     not have the same number. This requirement applies to every layer of nested
     repetitions.
 
+## Metavariable expressions
+
+Metavariable expressions in declarative macros provide expansions for information about metavariables that are otherwise not easily obtainable.
+
+| Expression                 | Meaning    |
+|----------------------------|------------|
+| `${ignore(ident)}`         | Binds `$ident` for repetition, but expands to nothing. |
+| `$$`                       | Expands to a single `$`, for removing ambiguity in recursive macro definitions. |
+
+### Ignore
+
+`${ignore(ident)}` repeats an expansion the same number of times a metavariable repeats without actually expanding the metavariable.
+
+```rust
+macro_rules! count {
+    ( $( $ignored_identifier:ident ),* ) => {{
+        0 $( + 1 ${ignore(ignored_identifier)} )*
+    }};
+}
+
+fn main() {
+    assert_eq!(count!(T, T, T), 3);
+}
+```
+
+The above snippet will expand to a sequence of numbers respecting the number of times that `ignored_identifier` repeats, resulting in a final `0 + 1 + 1 + 1` output.
+
+### Dollar-dollar ($$)
+
+`$$` expands to a single `$`.
+
+Since metavariable expressions always apply during the expansion of a macro, they cannot be used in recursive macro definitions and this is where `$$` expressions comes into play, i.e., `$$` can se used to resolve ambiguities in nested macros.
```

Comment:
```suggestion
Since metavariable expressions always apply during the expansion of a macro, they cannot be used in recursive macro definitions and this is where `$$` expressions comes into play, i.e., `$$` can be used to resolve ambiguities in nested macros.
```

Reply:
Thanks